addons/product/models/product_template.py

```diff
@@ -1108,3 +1108,16 @@ def get_import_templates(self):
             'label': _('Import Template for Products'),
             'template': '/product/static/xls/product_template.xls'
         }]
+
+    @api.multi
+    def _compute_message_attachment_count(self):
+        super(ProductTemplate, self)._compute_message_attachment_count()
+        record = self.filtered(lambda x: len(x.product_variant_ids) == 1)
```

Comment:
```suggestion
        records = self.filtered(lambda x: len(x.product_variant_ids) == 1)
```
Rename the variable here, as this is not always a singleton. And you need to handle those multiple records properly, as the next lines will crash.
(Easy to test: add `message_attachment_count` in the kanban view)